src/v/raft/consensus.cc

```diff
@@ -1383,10 +1381,32 @@ ss::future<> consensus::do_start() {
           "Configuration manager started: {}",
           _configuration_manager);
 
-        co_await _offset_translator.start(
-          offset_translator::must_reset{initial_state});
-
-        co_await _snapshot_lock.with([this] { return hydrate_snapshot(); });
+        std::optional<storage::truncate_prefix_config> start_truncate_cfg;
+        auto metadata = co_await read_snapshot_metadata();
+        if (metadata.has_value()) {
+            load_from_metadata(metadata.value());
+            co_await _configuration_manager.add(
+              _last_snapshot_index, std::move(metadata->latest_configuration));
+            _probe->configuration_update();
+
+            auto delta = delta_for_truncation(metadata.value());
+            if (delta.has_value()) {
+                start_truncate_cfg = storage::truncate_prefix_config(
+                  model::next_offset(_last_snapshot_index),
+                  _scheduling.default_iopc,
+                  delta);
+
+                _flushed_offset = std::max(
+                  _last_snapshot_index, _flushed_offset);
+                co_await _configuration_manager.prefix_truncate(
+                  _last_snapshot_index);
+            }
+            _snapshot_size = co_await _snapshot_mgr.get_snapshot_size();
+        }
```

Comment:
is this new code being added what had been done in hydrate snapshot?

Reply:
Yeah, I can try separating out a refactoring commit and maybe that'll make it clearer

The difference here between hydrate_snapshot() is that this is calling `start()` (which truncates and syncs the offset translator with the log) instead of just `truncate()`

Reply:
Pulled out a refactoring PR. Hopefully it's clearer now what the explicit differences in hydration are now